interfaces/devdoc/file_util_linux.md

```diff
@@ -0,0 +1,67 @@
+#define INVALID_HANDLE_VALUE            NULL
+
+typedef void *HANDLE;
+typedef void *LPSECURITY_ATTRIBUTES;
+
+MOCKABLE_FUNCTION(, HANDLE, ebs_create_file, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+MOCKABLE_FUNCTION(, bool, ebs_close_handle, HANDLE, handle_input);
+```
+
+### ebs_create_file
+
+```c
+MOCKABLE_FUNCTION(, HANDLE, ebs_create_file, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+```
+
+`ebs_create_file` implements the creating file function in Linux. Uses the `open` function in C for Linux.
+
+**SRS_FILE_UTIL_LINUX_09_001: [** If the `full_file_name` input is either empty or `NULL`, `ebs_create_file` shall return an `INVALID_HANDLE_VALUE`. **]**
+
+**SRS_FILE_UTIL_LINUX_09_002: [** `ebs_create_file` shall allocate memory for the file handle. ebs_create_file will succeed and return a non-`NULL` value.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_003: [** If memory allocation for `result` fails, `ebs_create_file` shall return an `INVALID_HANDLE_VALUE`. **]**
+
+**SRS_FILE_UTIL_LINUX_09_004: [** If `dwDesiredAccess` is `GENERIC_READ`, `ebs_create_file` will call `open` with `O_RDONLY` and shall return a file handle for read only.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_005: [** If `dwDesiredAcces` is `GENERIC_WRITE`, `ebs_create_file` will call `open` with `O_WRONLY` and shall return a file handle for write only.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_006: [** If `dwDesiredAccess` is `GENERIC_ALL` or `GENERIC_READ&GENERIC_WRITE`, `ebs_create_file` will call `open` with `O_RDWR` and shall return a file handle for read and write.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_014: [** If `creation_disposition` is `CREATE_ALWAYS` or `OPEN_ALWAYS`, `ebs_create_file` will call `open` with `O_CREAT` and shall either create a new file handle if the specificied pathname exists and return it or return an existing file handle.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_015: [** If `creation_disposition` is `CREATE_NEW`, `ebs_create_file` will call `open` with `O_CREAT|O_EXCL` and shall return a new file handle if the file doesn't already exist.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_016: [** If `creation_disposition` is `CREATE_NEW` and the file already exists, `ebs_create_file` will fail and return `INVALID_HANDLE_VALUE`.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_017: [** If creation_disposition is `TRUNCATE_EXISTING`, `ebs_create_file` will call `open` with `O_TRUNC` and shall return a file handle who's size has been truncated to zero bytes.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_007: [** If `open` returns a -1, `ebs_create_file` shall free memory and return an `INVALID_HANDLE_VALUE`.  **]**
```

Comment:
>SRS_FILE_UTIL_LINUX_09_007 [](http://example.com/codeflow?start=2&length=26)

It sounds like this is redundant with 09_008 below (or, there is nothing special about "open returns -1" as it is just "any failure"). Sometimes we have specific specs for failures if they are certain return values we handle as error, but in this case I would say that it is equivalent to "open failed".